internal/subshell/subshell.go

```diff
@@ -84,10 +86,20 @@ func New() SubShell {
 		}
 	}
 
-	logging.Debug("Detected SHELL: %s", binary)
+	// try to find the binary on the PATH
+	binaryPath, err := exec.LookPath(binary)
+	if err == nil {
+		// if we found it, resolve all symlinks, for many Linux distributions the SHELL is "sh" but symlinked to a different default shell like bash or zsh
+		resolved, err := fileutils.ResolvePath(binaryPath)
+		if err != nil {
+			logging.Debug("Failed to resolve path to shell binary %s: %v", binaryPath, err)
+		}
+		binary = resolved
```

Comment:
This should probably be in an `else`, since the value of `resolved` is not guaranteed if `err == nil`.

Reply:
Good point.